### PR TITLE
Xamlc compiled trigger

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

Based on #622

Compiled version of Trigger iValueProvider

this was already covered by existing tests

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/Interactivity/PropertyCondition.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public sealed class PropertyCondition : Condition, IValueProvider
 
 		public PropertyCondition()
 		{
-			_stateProperty = BindableProperty.CreateAttached("State", typeof(bool), typeof(DataTrigger), false, propertyChanged: OnStatePropertyChanged);
+			_stateProperty = BindableProperty.CreateAttached("State", typeof(bool), typeof(PropertyCondition), false, propertyChanged: OnStatePropertyChanged);
```
Comment (samhouts) [inline Xamarin.Forms.Core/Interactivity/PropertyCondition.cs]: That seems significant.

File: Xamarin.Forms.Build.Tasks/CompiledValueProviders/TriggerValueProvider.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+			var bpRef = (new BindablePropertyConverter()).GetBindablePropertyFieldReference((string)bpNode.Value, module, bpNode);
+
+			TypeReference _;
+			var setValueRef = module.Import(module.Import(typeof(Trigger)).GetProperty(p => p.Name == "Value", out _).SetMethod);
+
+			//push the setter
+			yield return Instruction.Create(OpCodes.Ldloc, vardefref.VariableDefinition);
+
+			//push the value
+			foreach (var instruction in ((ValueNode)valueNode).PushConvertedValue(context, bpRef, valueNode.PushServiceProvider(context, bpRef: bpRef), true, false))
```
Comment (samhouts) [inline Xamarin.Forms.Build.Tasks/CompiledValueProviders/TriggerValueProvider.cs]: Again, if you're in here, might be nice to have named parameters on the `bool`s.

File: Xamarin.Forms.Core/Interactivity/PropertyCondition.cs (hunk 1 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Interactivity/PropertyCondition.cs]: actually, it is not. AT ALL

Comment (samhouts) [inline Xamarin.Forms.Core/Interactivity/PropertyCondition.cs]: 👍

File: Xamarin.Forms.Build.Tasks/CompiledValueProviders/TriggerValueProvider.cs (hunk 2 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/CompiledValueProviders/TriggerValueProvider.cs]: I'm planning to drop that code anyway

**Pull request merged**